osquery/tables/system/darwin/signature.mm

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/filesystem.hpp>
+#include <Cocoa/Cocoa.h>                /* For NSAppKitVersionNumber */
+#include <Foundation/Foundation.h>
+#include <Security/CodeSigning.h>
+
+#include <osquery/core.h>
+#include <osquery/core/conversions.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+void genSignatureForFile(const std::string& path,
+                         QueryData& results) {
+  Row r;
+  OSStatus result = 0;
+  CFStringRef ident = NULL;
+  CFDictionaryRef codeInfo = NULL;
+  SecStaticCodeRef staticCode = NULL;
+  SecCSFlags csFlags = kSecCSDefaultFlags;
+
+  // Defaults
+  r["path"] = path;
+  r["signed"] = INTEGER(0);
+  r["identifier"] = "";
+
+  // Create a URL that points to this file.
+  auto url = (__bridge CFURLRef)[NSURL fileURLWithPath:@(path.c_str())];
+  if (url == nullptr) {
+    VLOG(1) << "Could not create URL from file: " << path;
+    goto finally;
+  }
+
+  // Create the static code object.
+  result = SecStaticCodeCreateWithPath(url, kSecCSDefaultFlags, &staticCode);
+  if (result != 0) {
+    VLOG(1) << "Could not create static code object for file: " << path;
+    goto finally;
+  }
+
+  // Set up the flags - some of them aren't present on 10.8.
+  if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_8) {
+    csFlags = kSecCSDefaultFlags | kSecCSCheckNestedCode;
+  }
+  else {
+    csFlags = kSecCSBasicValidateOnly;
+  }
+
+  // Actually validate.
+  result = SecStaticCodeCheckValidityWithErrors(staticCode, csFlags, NULL, NULL);
+  if (result != 0) {
+    // If this errors, then we either don't have a signature, or it's malformed.
+    VLOG(2) << "Static code validity check failed for file: " << path;
```

Comment:
Use `VLOG(1)` for everything in tables, the binaries take verbose level control away from the user and turn it into a default=WARNING/ERROR, verbose=VLOG(0-3)/INFO/WARNING/ERROR. There's also a seldom-used `TLOG`, that does not take an argument, which was designed to be the verbose logger within table implementations. :P